core/shared/src/test/scala/cats/parse/ErrorShowTest.scala

```diff
@@ -0,0 +1,182 @@
+package cats.parse
+
+import cats.implicits._
+
+import Parser._
+import Numbers.digits
+
+class ErrorShowTest extends munit.ScalaCheckSuite {
+
+  def error(parser: Parser0[Any], input: String, expected: String)(implicit
+      loc: munit.Location
+  ): Unit = {
+    test(input) {
+      parser
+        .parseAll(input)
+        .fold(
+          e =>
+            assert(
+              e.show == expected,
+              s"""|obtained:
+              |--------
+              |${e.show}
+              |--------
+              |
+              |expected:
+              |$expected""".stripMargin
+            ),
+          _ => fail("should not parse")
+        )
+    }
+  }
+
+  val ok = string("ok")
+  val nl = string("\n")
+  val lx = string("l") ~ digits
+  val lxOk = ((lx | ok) ~ nl)
+
+  // # Expectations:
+  // OneOfStr
+  error(
+    string("foo") | string("bar") | string("baz"),
+    "ko",
+    s"""|ko
+        |^
+        |* one of: {"bar", "baz", "foo"}""".stripMargin
+  )
+
+  // InRange
+  error(
+    charIn(List('a', 'c', 'x', 'y')),
+    "ko",
+    """|ko
+       |^
+       |* is: a
+       |* is: c
+       |* in range: [x, y]""".stripMargin
+  )
+
+  // StartOfString
+  error(
+    ok ~ start,
+    "ok",
+    """|ok
+       |  ^
+       |* start of string""".stripMargin
+  )
+
+  // EndOfString
+  error(
+    lx.repExactlyAs[Int](2),
+    "l1l2l3".stripMargin,
+    """|l1l2l3
+       |    ^
+       |* end of string, length = 6""".stripMargin
+  )
+
+  // Length
+  error(
+    ok | length(2),
+    "aaaaaaaaaaaaaaaaaaa",
+    """|aaaaaaaaaaaaaaaaaaa
+       |  ^
+       |* end of string, length = 19""".stripMargin
+  ) // this does repport the correct error IMO
```

Comment:
@johnynek do you know how can I trigger this Error?

Reply:
did you mean "this does *not* report"?

I would say something like `expected end of string but 19 characters remaining`.

This issue has a discussion about `|` hiding error messages:

https://github.com/typelevel/cats-parse/issues/382

Reply:
ah I think I figured it out, length is only trigger when the output is smaller

ex: `length(2).parseAll("a")`